src/core/signer.rs

```diff
@@ -189,7 +196,16 @@ impl Signer {
         };
 
         let sig = sign::sign(&self.code(), &self.raw(), ser)?;
-        Siger::new(None, Some(index), ondex, Some(code), Some(&sig), None, None, None)
+        Siger::new(
```

Comment:
Cigar gets the verfer when instantiated, so why not Siger? This didn't break any tests so I assume it's okay